core/shared/src/main/scala/zio/ConfigProvider.scala

```diff
@@ -492,8 +494,22 @@ object ConfigProvider {
                            .flatMap(set => indicesFrom(set))
 
               values <-
-                if (indices.isEmpty) loop(prefix, config, split = true).map(Chunk(_))
-                else
+                if (indices.isEmpty) {
+                  flat
+                    .load(prefix, Config.Text, false)
+                    .either
+                    .flatMap {
+                      case Left(_) =>
+                        loop(prefix, config, split = true).map(Chunk(_))
+
+                      case Right(chunk) =>
+                        if (chunk.size == 1 && (chunk.headOption.contains("<nil>")))
```

Comment:
Usually only libraries like zio-config (or advanced) would ever need to encode emptiness using `nil`. 

Users who prefer to write flat-structure hardly end up  doing and can prefer to use `""` as before. 

We can remove this pre-requisite in zio-3.x, where `ConfigPath` is the default behaviour